src/storage.cc

```diff
@@ -420,6 +431,15 @@ rocksdb::Status Storage::Write(const rocksdb::WriteOptions &options, rocksdb::Wr
   auto s = db_->Write(options, updates);
   if (!s.ok()) return s;
 
+  if (updates->Count() != 0) {
```

Comment:
is it ok that doesn't emit the commit event when the semi-sync was disabled? 

BTW: need an indent inside the if block

Reply:
the event is going to be blocked at the begin of ReplSemiSyncMaster::CommitTrx if semi sync is disabled， moreover before event return， program always try to wake up feedslavethread to get latest data from wal regardless of semi sync status.  i didn't check semi sync status on storage because config setting is not final result. actually i check if the semi sync is working according to  semi_sync_enabled_ field of ReplSemiSyncMaster and i think storage shouldn't reference to ReplSemiSyncMaster object

Reply:
> is it ok that doesn't emit the commit event when the semi-sync was disabled?
> 
> BTW: need an indent inside the if block

the event is going to be blocked at the begin of ReplSemiSyncMaster::CommitTrx if semi sync is disabled， moreover before event return， program always try to wake up feedslavethread to get latest data from wal regardless of semi sync status. i didn't check semi sync status on storage because config setting is not final result. actually i check if the semi sync is working according to semi_sync_enabled_ field of ReplSemiSyncMaster and i think storage shouldn't reference to ReplSemiSyncMaster object